share/sysbuild/image_config.cmake

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2023 Nordic Semiconductor
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# This sysbuild CMake file sets the sysbuild controlled settings as properties
+# on Zephyr images.
+
+get_target_property(image_board ${image} BOARD)
+if((NOT image_board) OR ("${image_BOARD}" STREQUAL "${BOARD}"))
```

Comment:
This doesn't work for boards which have a default revision.
For example, if I build for `nrf9160dk_nrf9160`, then: `image_board == "nrf9160dk_nrf9160@0.7.0" != BOARD`

Reply:
good catch 👍 
This happens because the BOARD here is always without revision because [boards.cmake](https://github.com/zephyrproject-rtos/zephyr/blob/main/cmake/modules/boards.cmake) has split the setting, whereas the `image_BOARD` is the raw value.

Will take a look.

Reply:
This is still an issue.

One possible solution is to cherry-pick https://github.com/zephyrproject-rtos/zephyr/pull/58174/commits/fa1a137003cfadc54211573ef8dd5afb757a16fa, if you'd like (from my sysbuild grouping PR).

Reply:
> This is still an issue (doesn't work for boards with a default revision).

I know, which is why I didn't mark it as fixed 😉 
When looking at it I discovered another flaw wrt. BOARD that I wanted to address first.
Part of it which you have also discovered, but there is also the possibility of a user specifying a dedicated BOARD for an image, such as `-DBOARD=myboard -Dmcuboot_BOARD=myboard@<revision>` where user wants to build mcuboot for specific revision even though the main app could be targeting another revision.

In such a case, the revision will be part of the image `BOARD` setting.

(edit: comment has been moved)